resin/models/build.py

```diff
@@ -0,0 +1,81 @@
+from ..base_request import BaseRequest
+from ..settings import Settings
+from ..token import Token
+from .. import exceptions
+
+
+class Build(object):
+    """
+    This class implements model for build for Resin Python SDK.
+
+    """
+
+    def __init__(self):
+        self.base_request = BaseRequest()
+        self.settings = Settings()
+        self.token = Token()
+
+    def get(self, id):
+        """
+        Get a specific build.
+
+        Args:
+            id (str): build id.
+
+        Returns:
+            dict: build info.
+
+        Raises:
+            BuildNotFound: if build couldn't be found.
+
+        """
+
+        params = {
+            'filter': 'id',
+            'eq': id
+        }
+        build = self.base_request.request(
+            'build', 'GET', params=params,
+            endpoint=self.settings.get('pine_endpoint')
+        )
+        if build:
+            return build
+        else:
+            raise exceptions.BuildNotFound(id)
+
+    def get_all_by_application(self, app_id, build_log=None):
+        """
+        Get a specific build.
+
+        Args:
+            app_id (str): application id.
+            build_log (bool): True if user wants to include build logs in build info, False otherwise.
+
+        Returns:
+            list: list of build info.
+
+        Raises:
+            BuildNotFound: if build couldn't be found.
+
+        """
+
+        params = {
+            'filter': 'application',
+            'eq': app_id
+        }
+        full_builds = self.base_request.request(
+            'build', 'GET', params=params,
+            endpoint=self.settings.get('pine_endpoint')
+        )
+        builds = []
+        build_info = [
+            'id', 'created_at', 'status',
+            'push_timestamp', 'end_timestamp',
+            'start_timestamp', 'project_type',
+            'commit_hash', 'message'
+        ]
+        if build_log:
+            build_info.append('log')
```

Comment:
The JS code only lists all the fields because it has to for the pine query. You don't have to, so you could avoid that and avoid duplicating the list of fields in the model. It doesn't make much difference if you're adding pine options really soon, but if not, or if you want this to be clearer in the meantime, you could replace everything after `full_builds` with:

```python
builds = full_builds['d']
if include_logs:
  return builds
else:
  return { k: v for build in builds for k, v in build.items() if k is not 'log'}
```